docs/developer-handbook.md

```diff
@@ -0,0 +1,282 @@
+  - [Subject](#subject)
+  - [Body](#body)
+  - [Footer](#footer)
+  - [Examples](#examples)
+- [FAQ](#faq)
+    - [How do I install a dependency?](#how-do-i-install-a-dependency)
+    - [CircleCI is failing saying that it cannot find a dependency in offline mode](#circleci-is-failing-saying-that-it-cannot-find-a-dependency-in-offline-mode)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- prettier-ignore-end -->
+
+## Getting started
+
+Carbon is built using a collection of packages all built in the same git
+repository. You might have heard this setup described as a
+[monorepo](https://en.wikipedia.org/wiki/Monorepo).
+
+As a result, we use two pieces of tooling to help us managing installing
+dependencies and publishing our packages. These include:
+
+- [Yarn workspaces](https://yarnpkg.com/lang/en/docs/workspaces/) for handling
+  dependencies across all packages
+- [Lerna](https://lernajs.io/) for publishing packages, tagging versions, and
+  more
+
+In order for you to install all the dependencies in this project, you'll need to
+[install Yarn](https://yarnpkg.com/en/docs/install) and run the following
+command in your terminal:
+
+```bash
+yarn install
+```
+
+This will install all of the dependencies for every package in our project. In
+addition, it allows us to link between packages that we are developing.
+
+This strategy is particularly useful during development, and tooling like Lerna
+will pick up on when packages are linked in this way and will automatically
+update versions when publishing new versions of packages.
+
+Next up, you'll most likely want to build all of the package files so that
+things don't fail while you are working on a package. To do this, you can run
+the following command:
+
+```bash
+yarn build
+```
+
+Afterwards, you should be good to go! For more information about how we handle
+dependencies, definitely take a look at our write-up
+[here](#dependency-management)
+
+## Common tasks
+
+While working on Carbon, here are some of the top-level tasks that you might
+want to run:
+
+| Command                           | Usage                                                                                                         |
+| --------------------------------- | ------------------------------------------------------------------------------------------------------------- |
+| `yarn build`                      | Uses `lerna` to run the `build` script in each package                                                        |
+| `yarn clean`                      | Resets the state of the project by removing all `node_modules` and running the `clean` script in each package |
+| `yarn doctoc`                     | Runs `doctoc` on all files in the `doctoc` directory                                                          |
+| `yarn format`, `yarn format:diff` | Format files using prettier, check if files have been formatted                                               |
```

Comment:
I've had to run yarn lint as well in order to get commits to be successful. Might be worth mentioning here since it can be a commit blocker.

Reply:
Agreed!